coderd/workspaceagentsrpc.go

```diff
@@ -156,7 +157,7 @@ func (api *API) workspaceAgentRPC(rw http.ResponseWriter, r *http.Request) {
 	ctx = tailnet.WithStreamID(ctx, streamID)
 	ctx = agentapi.WithAPIVersion(ctx, version)
 	err = agentAPI.Serve(ctx, mux)
-	if err != nil {
+	if err != nil && !xerrors.Is(err, yamux.ErrSessionShutdown) && !xerrors.Is(err, io.EOF) {
 		logger.Warn(ctx, "workspace agent RPC listen error", slog.Error(err))
 		_ = conn.Close(websocket.StatusInternalError, err.Error())
```

Comment:
Should you still call close even on these errors?

Reply:
I don't think so.  The netConn is closed via a `defer`, which closes the underlying websocket --- the only difference is that it will send `NormalClosure` rather than `InternalError`, which seems fine for these error types.